src/main/java/com/uber/jenkins/phabricator/BuildResultProcessor.java

```diff
@@ -0,0 +1,205 @@
+        this.logger = logger;
+        this.diff = diff;
+        this.diffClient = diffClient;
+        this.phid = phid;
+        this.buildUrl = buildUrl;
+
+        this.buildResult = build.getResult();
+        this.workspace = build.getWorkspace();
+
+        this.commentAction = "none";
+        this.commenter = new CommentBuilder(logger, build.getResult(), coverageResult, buildUrl);
+        this.runHarbormaster = !CommonUtils.isBlank(phid);
+    }
+
+    public void processParentCoverage(UberallsClient uberalls) {
+        // First add in info about the change in coverage, if applicable
+        if (commenter.hasCoverageAvailable()) {
+            if (uberalls.isConfigured()) {
+                commenter.processParentCoverage(uberalls.getParentCoverage(diff.getBaseCommit()), diff.getBaseCommit(), diff.getBranch());
+            } else {
+                logger.info(LOGGING_TAG, "No Uberalls backend configured, skipping...");
+            }
+        } else {
+            logger.info(LOGGING_TAG, "No line coverage found, skipping...");
+        }
+
+    }
+
+    public void processBuildResult(boolean commentOnSuccess, boolean commentWithConsoleLinkOnFailure) {
+        commenter.processBuildResult(commentOnSuccess, commentWithConsoleLinkOnFailure, runHarbormaster);
+    }
+
+    public void processRemoteComment(String commentFile, String commentSize) {
+        RemoteCommentFetcher commentFetcher = new RemoteCommentFetcher(workspace, logger, commentFile, commentSize);
+        try {
+            String customComment = commentFetcher.getRemoteComment();
+            commenter.addUserComment(customComment);
+        } catch (InterruptedException e) {
+            e.printStackTrace(logger.getStream());
+        } catch (IOException e) {
+            e.printStackTrace(logger.getStream());
+        }
+    }
+
+    public void sendComment(boolean commentWithConsoleLinkOnFailure) {
+        if (!commenter.hasComment()) {
+            return;
+        }
+
+        if (commentWithConsoleLinkOnFailure && buildResult.isWorseOrEqualTo(hudson.model.Result.UNSTABLE)) {
+            commenter.addBuildFailureMessage();
+        } else {
+            commenter.addBuildLink();
+        }
+
+        new PostCommentTask(logger, diffClient, diff.getRevisionID(false), commenter.getComment(), commentAction).run();
+    }
+
+    public boolean processHarbormaster() {
+        final boolean harbormasterSuccess = buildResult.isBetterOrEqualTo(Result.SUCCESS);
+
+        if (runHarbormaster) {
+            logger.info("harbormaster", "Sending Harbormaster BUILD_URL via PHID: " + phid);
+            Task.Result sendUriResult = new SendHarbormasterUriTask(logger, diffClient, phid, buildUrl).run();
+
+            if (sendUriResult != Task.Result.SUCCESS) {
+                logger.info(LOGGING_TAG, "Unable to send BUILD_URL to Harbormaster");
+            }
+
+            if (getCoverage() != null) {
```

Comment:
Why don't you just access the class variable here instead of making the method call?